gateway-ha/src/main/java/io/trino/gateway/ha/router/RoutingRulesManager.java

```
@@ -0,0 +1,84 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.router;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLParser;
+import io.trino.gateway.ha.config.RoutingRulesConfiguration;
+import io.trino.gateway.ha.domain.RoutingRules;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class RoutingRulesManager
+{
+    public List<RoutingRules> getRoutingRules(RoutingRulesConfiguration configuration, ObjectMapper yamlReader)
+            throws IOException
+    {
+        String content = null;
+        try {
+            String rulesConfigPath = configuration.getRulesConfigPath();
+            content = new String(Files.readAllBytes(Paths.get(rulesConfigPath)), UTF_8);
+            YAMLParser parser = new YAMLFactory().createParser(content);
+            List<RoutingRules> routingRulesList = new ArrayList<>();
+            while (parser.nextToken() != null) {
+                RoutingRules routingRules = yamlReader.readValue(parser, RoutingRules.class);
+                routingRulesList.add(routingRules);
+            }
+            return routingRulesList;
+        }
+        catch (IOException e) {
+            throw new IOException(e);
```

Please rethrow an appropriate exception.